tests/test_models.py

```diff
@@ -77,7 +78,7 @@ def test_zone_add_record_unknown_type():
 
 def test_zone_add_record_no_content():
     with localzone.manage(ZONEFILE, ORIGIN) as z:
-        with pytest.raises(AttributeError):
+        with pytest.raises((AttributeError, DNSSyntaxError)):
```

Comment:
I am not sure, why this one comes out as SyntaxError:

```
[    5s] _______________________ test_zone_add_record_no_content ________________________
[    5s] 
[    5s] rdclass = <RdataClass.IN: 1>, rdtype = <RdataType.TXT: 16>, tok = None
[    5s] origin = <DNS name example.com.>, relativize = True, relativize_to = None
[    5s] idna_codec = None
[    5s] 
[    5s]     def from_text(rdclass, rdtype, tok, origin=None, relativize=True,
[    5s]                   relativize_to=None, idna_codec=None):
[    5s]         """Build an rdata object from text format.
[    5s]     
[    5s]         This function attempts to dynamically load a class which
[    5s]         implements the specified rdata class and type.  If there is no
[    5s]         class-and-type-specific implementation, the GenericRdata class
[    5s]         is used.
[    5s]     
[    5s]         Once a class is chosen, its from_text() class method is called
[    5s]         with the parameters to this function.
[    5s]     
[    5s]         If *tok* is a ``str``, then a tokenizer is created and the string
[    5s]         is used as its input.
[    5s]     
[    5s]         *rdclass*, an ``int``, the rdataclass.
[    5s]     
[    5s]         *rdtype*, an ``int``, the rdatatype.
[    5s]     
[    5s]         *tok*, a ``dns.tokenizer.Tokenizer`` or a ``str``.
[    5s]     
[    5s]         *origin*, a ``dns.name.Name`` (or ``None``), the
[    5s]         origin to use for relative names.
[    5s]     
[    5s]         *relativize*, a ``bool``.  If true, name will be relativized.
[    5s]     
[    5s]         *relativize_to*, a ``dns.name.Name`` (or ``None``), the origin to use
[    5s]         when relativizing names.  If not set, the *origin* value will be used.
[    5s]     
[    5s]         *idna_codec*, a ``dns.name.IDNACodec``, specifies the IDNA
[    5s]         encoder/decoder to use if a tokenizer needs to be created.  If
[    5s]         ``None``, the default IDNA 2003 encoder/decoder is used.  If a
[    5s]         tokenizer is not created, then the codec associated with the tokenizer
[    5s]         is the one that is used.
[    5s]     
[    5s]         Returns an instance of the chosen Rdata subclass.
[    5s]     
[    5s]         """
[    5s]         if isinstance(tok, str):
[    5s]             tok = dns.tokenizer.Tokenizer(tok, idna_codec=idna_codec)
[    5s]         rdclass = dns.rdataclass.RdataClass.make(rdclass)
[    5s]         rdtype = dns.rdatatype.RdataType.make(rdtype)
[    5s]         cls = get_rdata_class(rdclass, rdtype)
[    5s]         with dns.exception.ExceptionWrapper(dns.exception.SyntaxError):
[    5s]             rdata = None
[    5s]             if cls != GenericRdata:
[    5s]                 # peek at first token
[    5s] >               token = tok.get()
[    5s] E               AttributeError: 'NoneType' object has no attribute 'get'
[    5s] 
[    5s] /usr/lib/python3.6/site-packages/dns/rdata.py:589: AttributeError
[    5s] 
[    5s] The above exception was the direct cause of the following exception:
[    5s] 
[    5s]     def test_zone_add_record_no_content():
[    5s]         with localzone.manage(ZONEFILE, ORIGIN) as z:
[    5s]             with pytest.raises(AttributeError):
[    5s] >               z.add_record("test", "txt", None)
[    5s] 
[    5s] tests/test_models.py:81: 
[    5s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[    5s] localzone/models.py:195: in add_record
[    5s]     rdata = dns.rdata.from_text(rdclass, rdtype, content, origin=self.origin)
[    5s] /usr/lib/python3.6/site-packages/dns/rdata.py:618: in from_text
[    5s]     return rdata
[    5s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[    5s] 
[    5s] self = <dns.exception.ExceptionWrapper object at 0x7f06610b8550>
[    5s] exc_type = <class 'AttributeError'>
[    5s] exc_val = AttributeError("'NoneType' object has no attribute 'get'",)
[    5s] exc_tb = <traceback object at 0x7f0661170dc8>
[    5s] 
[    5s]     def __exit__(self, exc_type, exc_val, exc_tb):
[    5s]         if exc_type is not None and not isinstance(exc_val,
[    5s]                                                    self.exception_class):
[    5s] >           raise self.exception_class(str(exc_val)) from exc_val
[    5s] E           dns.exception.SyntaxError: 'NoneType' object has no attribute 'get'
[    5s] 
[    5s] /usr/lib/python3.6/site-packages/dns/exception.py:141: SyntaxError

```

Reply:
Likely a more strict adherence by dnspython to RFC 1035 which reads "[o]ne or more <character-string>s". Thanks for fixing!